src/lib/learn-client/api/progress/formatting.ts

```diff
@@ -0,0 +1,40 @@
+import {
+	TutorialProgressLabel,
+	TutorialProgressPercent,
+} from 'lib/learn-client/types'
+
+/**
+ * Convert from API's complete_percent to the TutorialProgressLabel enum.
+ *
+ * This is necessary as in the context of Dev Dot, we want the label,
+ * but in the context of the Learn API, we want complete_percent.
+ */
+export function progressPercentToLabel(
```

Comment:
Formatting utilities to convert between the API format (a limited set of percentage values), and the format we want to display and reason around on Dev Dot (`visited`, `in_progress`, or `completed`).

This is largely informed by existing behaviour in `hashicorp/learn` - while it's technically possible for the API to hold a variety of percentage values, we only ever create or update with `0`, `50`, or `100` (corresponding to `visited`, `in_progress`, and `completed`).

Reply:
@kendallstrautman do you have more context on why we only create/update Tutorial:Collection progress with 0, 50, or 100? Doesn't this mean that a user will have to get 50% of the way through a tutorial before their progress is set as "in progress"?

Reply:
@ashleemboyer With reference to both the existing `hashicorp/learn` code and to this PR, the `0`, `50`, and `100` percent values are not currently tied to scroll progress in a strict sense. Instead we use `start` and `end` markers. 
- when a page is loaded, we set `0%` in the DB (`visited` in the front-end)
- when the page is scrolled enough that the `start` marker is out of view, we set `50%` in the DB (`in_progress`)
- when the page is scrolled enough that the `end` marker is in view, we set `100%` in the DB (`complete`)

The `in_progress` state has been tied to `50%` previously, so the intent in this PR is to maintain that mapping.

Let me know if that clarifies. This might also related to the layer of opinion built into the `react-query` layer.

Reply:
^ zach explained it really well. From the application perspective, we are only concerned with four states: 'no progress' (record doesn't exist yet), 'visited' (0), 'in progress' (50 - but this is an arbitrary number beween 0-100), & 'complete' (100).

The percentages in the API were created that way to make the system more flexible in case we wanted to capture more finite progress metrics later down the line.